### PR TITLE
12286 Implement benchmark for callLater() and reactor.runUntilCurrent() and speed them up

### LINKED ISSUE #12286: Implement benchmark for callLater and reactor.runUntilCurrent(), and speed them up
It looks like there are opportunities to speed up scheduled events.

### PR DESCRIPTION
**Pull request by itamarst**
## Scope and purpose

Fixes #12286 

The benchmark is about 20% faster on my computer after optimizations. Only a small part of a program's execution, of course, but every little bit adds up.

### CODE REVIEW

File: src/twisted/newsfragments/12286.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+The reactor now will use a little CPU when events have been scheduled with callLater().
```
Comment (adiroiban) [inline src/twisted/newsfragments/12286.feature]: just looking at this 

```suggestion
The reactor now will use a little less CPU when events have been scheduled with callLater().
```

File: benchmarks/test_calllater.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import pytest
+
+from twisted.internet.base import ReactorBase
+
+
+class Reactor(ReactorBase):
+    """
+    Implement minimal methods a subclass needs.
+    """
```
Comment (adiroiban) [inline benchmarks/test_calllater.py]: I think that this need more documentation.

My initial question was: 
why not use [t.i.task.Clock](https://docs.twistedmatrix.com/en/stable/api/twisted.internet.task.Clock.html) ?

```suggestion
class ReactorWithRiggedScheduler(ReactorBase):
    """
    Allows controlling the passed time while testing the C{ReactorBase}.
    """
```

**Review by adiroiban — APPROVED** [review]
Thanks. Looks good. Only a minor comment.

File: benchmarks/test_calllater.py (hunk 2 above)
Comment (itamarst) [reply benchmarks/test_calllater.py]: It's not `Clock` because that would be benchmarking `Clock`, which is a completely different implementation.

**Pull request merged**